manifests/params.pp

```diff
@@ -38,6 +38,17 @@
       $rabbitmq_home    = '/var/rabbitmq'
       $plugin_dir       = '/usr/local/lib/rabbitmq/plugins'
     }
+    'FreeBSD': {
+      $package_ensure   = 'installed'
+      $package_name     = 'rabbitmq'
+      $service_name     = 'rabbitmq'
+      $package_provider = 'pkgng'
+      $version          = '3.4.2'
```

Comment:
Also, keep in mind that while `$version` is used internally as things work now, it's ignored when using the system package provider, so it doesn't really matter in this case.